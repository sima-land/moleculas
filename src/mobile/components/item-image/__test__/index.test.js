import React, { Component } from 'react';
import { mount } from 'enzyme';
import ItemImage from '../index';
import { act } from 'react-dom/test-utils';

describe('<ItemImage />', () => {
  it('should render without props', () => {
    const wrapper = mount(<ItemImage />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render wish', () => {
    const wrapper = mount(<ItemImage wishProps={{}} hasWishButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render wish, no props', () => {
    const wrapper = mount(<ItemImage wishProps={undefined} hasWishButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with withBlur prop', () => {
    const wrapper = mount(<ItemImage withBlur />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render badges', () => {
    const wrapper = mount(
      <ItemImage badges={[
        {
          bgColor: '#b52ea8',
          fields: [
            {
              type: 'text',
              value: '-36%',
            },
          ],
          link: '/percent/',
          strokeColor: null,
          textColor: '#f2f2f2',
          title: null,
        },
      ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should onLoadImage() calls', () => {
    const origHTMLImageElementDescriptors = Object.getOwnPropertyDescriptors(HTMLImageElement);
    Object.defineProperty(HTMLImageElement.prototype, 'complete', {
      writable: true,
      value: true,
    });
    // eslint-disable-next-line require-jsdoc
    class Wrp extends Component {
      // eslint-disable-next-line require-jsdoc
      render () { return (
        <div>{this.props.children}</div>
      );}
    }
    const spy = jest.fn();
    const wrapper = mount(
      <Wrp><ItemImage onLoadImage={spy} src='https://via.placeholder.com/350x150' /></Wrp>
    );
    act(() => {
      document.querySelector('.image')?.load();
      wrapper.mount();
    });
    expect(spy).toHaveBeenCalled();
    Object.defineProperties(HTMLImageElement, origHTMLImageElementDescriptors);
  });
});
