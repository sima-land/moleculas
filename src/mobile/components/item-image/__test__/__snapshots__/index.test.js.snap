// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemImage /> should render badges 1`] = `
<ItemImage
  badges={
    Array [
      Object {
        "bgColor": "#b52ea8",
        "fields": Array [
          Object {
            "type": "text",
            "value": "-36%",
          },
        ],
        "link": "/percent/",
        "strokeColor": null,
        "textColor": "#f2f2f2",
        "title": null,
      },
    ]
  }
>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)>
        <a
          className="link brand-blue"
          data-testid="anchor"
        >
          <img
            className="image"
          />
        </a>
      </ForwardRef(Link)>
    </div>
    <div
      className="badges"
    >
      <FlagsList
        flags={
          Array [
            Object {
              "bgColor": "#b52ea8",
              "fields": Array [
                Object {
                  "type": "text",
                  "value": "-36%",
                },
              ],
              "link": "/percent/",
              "strokeColor": null,
              "textColor": "#f2f2f2",
              "title": null,
            },
          ]
        }
      >
        <div
          className="wrap"
        >
          <Badge
            bgColor="#b52ea8"
            className="item"
            fields={
              Array [
                Object {
                  "type": "text",
                  "value": "-36%",
                },
              ]
            }
            key="0"
            link="/percent/"
            strokeColor={null}
            textColor="#f2f2f2"
            title={null}
          >
            <span
              bgColor="#b52ea8"
              className="root item"
              data-testid="badge"
              link="/percent/"
              strokeColor={null}
              style={
                Object {
                  "--badge-color": undefined,
                }
              }
              textColor="#f2f2f2"
              title={null}
            >
              <span
                className="content"
              >
                -36%
              </span>
            </span>
          </Badge>
        </div>
      </FlagsList>
    </div>
  </div>
</ItemImage>
`;

exports[`<ItemImage /> should render wish 1`] = `
<ItemImage
  hasWishButton={true}
  wishProps={Object {}}
>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)>
        <a
          className="link brand-blue"
          data-testid="anchor"
        >
          <img
            className="image"
          />
        </a>
      </ForwardRef(Link)>
    </div>
    <ForwardRef>
      <ForwardRef(SvgHeart)
        aria-label="Добавить в избранное"
        className="button"
        data-testid="favorite-button"
        height={24}
        role="button"
        width={24}
      >
        <svg
          aria-label="Добавить в избранное"
          className="button"
          data-testid="favorite-button"
          height={24}
          role="button"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.944 3.6h-.001c-2.7-1.879-5.978-1.165-7.943.793-1.964-1.958-5.241-2.683-7.943-.792C2.524 4.673 1.57 6.473 1.504 8.369c-.075 2.145.826 4.042 2.36 5.985 1.512 1.912 3.685 3.927 6.261 6.315l.086.08h.002l.098.091a2.466 2.466 0 003.371-.012l.107-.1.027-.025c2.602-2.407 4.795-4.436 6.316-6.361 1.534-1.941 2.435-3.838 2.364-5.982v-.001c-.065-1.886-1.02-3.686-2.552-4.758zM12 18.274c-2.278-2.105-4.14-3.84-5.45-5.401-1.358-1.62-2.047-2.968-2.047-4.261 0-1.774 1.285-3.07 2.999-3.07 1.055 0 2.103.553 2.72 1.39.38.516.996 1.017 1.78 1.017.789 0 1.406-.507 1.787-1.025.612-.831 1.657-1.382 2.71-1.382 1.713 0 2.998 1.296 2.998 3.07 0 1.293-.689 2.641-2.047 4.26-1.309 1.561-3.171 3.296-5.45 5.402z"
          />
        </svg>
      </ForwardRef(SvgHeart)>
    </ForwardRef>
  </div>
</ItemImage>
`;

exports[`<ItemImage /> should render wish, no props 1`] = `
<ItemImage
  hasWishButton={true}
>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)>
        <a
          className="link brand-blue"
          data-testid="anchor"
        >
          <img
            className="image"
          />
        </a>
      </ForwardRef(Link)>
    </div>
    <ForwardRef>
      <ForwardRef(SvgHeart)
        aria-label="Добавить в избранное"
        className="button"
        data-testid="favorite-button"
        height={24}
        role="button"
        width={24}
      >
        <svg
          aria-label="Добавить в избранное"
          className="button"
          data-testid="favorite-button"
          height={24}
          role="button"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.944 3.6h-.001c-2.7-1.879-5.978-1.165-7.943.793-1.964-1.958-5.241-2.683-7.943-.792C2.524 4.673 1.57 6.473 1.504 8.369c-.075 2.145.826 4.042 2.36 5.985 1.512 1.912 3.685 3.927 6.261 6.315l.086.08h.002l.098.091a2.466 2.466 0 003.371-.012l.107-.1.027-.025c2.602-2.407 4.795-4.436 6.316-6.361 1.534-1.941 2.435-3.838 2.364-5.982v-.001c-.065-1.886-1.02-3.686-2.552-4.758zM12 18.274c-2.278-2.105-4.14-3.84-5.45-5.401-1.358-1.62-2.047-2.968-2.047-4.261 0-1.774 1.285-3.07 2.999-3.07 1.055 0 2.103.553 2.72 1.39.38.516.996 1.017 1.78 1.017.789 0 1.406-.507 1.787-1.025.612-.831 1.657-1.382 2.71-1.382 1.713 0 2.998 1.296 2.998 3.07 0 1.293-.689 2.641-2.047 4.26-1.309 1.561-3.171 3.296-5.45 5.402z"
          />
        </svg>
      </ForwardRef(SvgHeart)>
    </ForwardRef>
  </div>
</ItemImage>
`;

exports[`<ItemImage /> should render with withBlur prop 1`] = `
<ItemImage
  withBlur={true}
>
  <div
    className="wrapper"
  >
    <div
      className="with-blur"
    >
      <ForwardRef(SvgEighteenPlus)
        height={124}
        width={124}
      >
        <svg
          height={124}
          viewBox="0 0 36 36"
          width={124}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M30.822 10.819h-1.438V6.746h-4.072V5.31h4.072V1.236h1.438V5.31h4.072v1.437h-4.072v4.073zm-10.418 4.505c.368.418 1.047.834 2.04 1.252.892-.38 1.523-.787 1.894-1.223.371-.437.557-.946.557-1.528 0-.638-.226-1.145-.678-1.522-.452-.376-1.075-.564-1.867-.564-.78 0-1.391.19-1.834.569-.443.38-.664.885-.664 1.517 0 .582.183 1.082.552 1.499zm-.365 3.941c-.418.49-.627 1.076-.627 1.76 0 .758.26 1.35.78 1.773.522.424 1.253.635 2.195.635.93 0 1.653-.22 2.17-.663.519-.443.778-1.05.778-1.822 0-.613-.243-1.158-.73-1.635-.487-.478-1.335-.94-2.545-1.39-.93.404-1.603.852-2.021 1.342zm-1.151-3.585a3.433 3.433 0 01-.562-1.921c0-1.005.367-1.807 1.1-2.404.733-.598 1.714-.897 2.943-.897 1.248 0 2.236.294 2.966.883.73.588 1.095 1.4 1.095 2.437a3.06 3.06 0 01-.627 1.869c-.418.562-1.085 1.075-2.003 1.536 1.11.537 1.9 1.102 2.368 1.693.468.592.702 1.276.702 2.054 0 1.15-.397 2.069-1.188 2.755-.793.686-1.879 1.03-3.257 1.03-1.46 0-2.583-.325-3.369-.973-.786-.648-1.179-1.566-1.179-2.755 0-1.587.955-2.824 2.864-3.71-.861-.492-1.479-1.025-1.853-1.597zm-.883 20.292C8.083 35.972.04 27.928.04 18.005.039 8.083 8.083.04 18.005.04c2.095 0 4.106.358 5.975 1.017.014.319.014 1.503.014 1.498-1.863-.728-3.867-1.078-5.989-1.078-9.129 0-16.529 7.4-16.529 16.53 0 9.128 7.4 16.528 16.53 16.528 9.128 0 16.529-7.4 16.529-16.529 0-2.116-.398-4.139-1.123-5.999.793.01 1.398.01 1.537.009a17.936 17.936 0 011.023 5.99c0 9.923-8.044 17.967-17.967 17.967zM9.487 14.575l-.823-1.081 3.621-2.836h1.31v13.866h-1.516v-9.883c0-.822.025-1.6.075-2.333a10.18 10.18 0 01-.44.418c-.162.145-.904.762-2.227 1.849z"
            fillRule="evenodd"
            opacity={0.3}
          />
        </svg>
      </ForwardRef(SvgEighteenPlus)>
    </div>
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)>
        <a
          className="link brand-blue"
          data-testid="anchor"
        >
          <div
            className="with-filter-blur"
          >
            <img
              className="image"
            />
          </div>
        </a>
      </ForwardRef(Link)>
    </div>
  </div>
</ItemImage>
`;

exports[`<ItemImage /> should render without props 1`] = `
<ItemImage>
  <div
    className="wrapper"
  >
    <div
      className="image-wrapper"
    >
      <ForwardRef(Link)>
        <a
          className="link brand-blue"
          data-testid="anchor"
        >
          <img
            className="image"
          />
        </a>
      </ForwardRef(Link)>
    </div>
  </div>
</ItemImage>
`;
